### PR TITLE
Issue #12807: Give better names to release actions and files

### LINKED ISSUE #12807: Give better names to release actions and files
Identified at https://github.com/checkstyle/checkstyle/issues/12453 ,

https://github.com/checkstyle/checkstyle/blob/master/.github/workflows/update-sources.yml
Action file name is called `update sources`, but internal name is `Push Release Notes`, which calls the shell file `update-releasenotesxml`.

Naming doesn't seem consistent between action file name, internal name, and script name. It will make debugging this confusing since there is no consistency.

I recommend we give all 3 similar names. It is fine if file names are shorter and internal name is longer, or vice versa, but there should be some consistency to easily link the 3 together.
For example, I think `update-releasenotes` and `Update Release Notes` works fine.

We may also want to consider all release scripts/actions starting with `r-` or `release-` similar to the action's internal name.

I am using the first one as an example for now, but all of the release process should be reviewed too.

current action / shell file:
```
update-sources.yml
  update-releasenotesxml.sh

maven-prepare.yml
  maven-prepare.sh

maven-release-perform.yml
  maven-release-perform.sh

new-milestone-and-issue-in-other-repo.yml
  close-create-milestone.sh
  creation-of-issue-in-other-repos.sh

publish-releasenotes-outside.yml
  update-github-page.sh

update-github-io.yml
  push-website-to-github-io.sh

upload-all-jar.yml
  upload-all-jar.sh

copy-site-to-sourceforge.yml
  copy-site-to-sourceforge.sh

publish-releasenotes-for-twitter.yml
  tweet-releasenotes.sh
```

### PR DESCRIPTION
**Pull request by stoyanK7**
Resolves #12807 

I did a release run to make sure everything is still connected.
https://github.com/stoyanK7/checkstyle/actions/runs/4418961016

Screenshot from 2023-03-14 19-57-25

### CODE REVIEW

**Comment by stoyanK7** [comment]
This PR maybe should be closed because https://github.com/checkstyle/checkstyle/issues/12807#issuecomment-1465224843 proposes different changes. I just noticed.

**Comment by romani** [comment]
Let's wait for final approval in issue, we can do one more update after approval

File: .github/workflows/release-publish-releasenotes-twitter.yml [hunk 1]
```diff
@@ -2,8 +2,8 @@
 # GitHub Action to Publish Tweet of release notes.
 #
 #############################################################################
-name: "R: Publish Tweet"
-run-name: "R: Publish Tweet ${{ inputs.version }}"
+name: "R: Publish Releasenotes Twitter"
+run-name: "R: Publish Releasenotes Twitter ${{ inputs.version }}"
```
Comment (stoyanK7) [inline .github/workflows/release-publish-releasenotes-twitter.yml:6]: I should change this to `R: Publish Release Notes Twitter` to stay consistent with `R: Update xdoc with Release Notes`

**Comment by romani** [comment]
`release-create-maven-settings-xml.sh`

File: .github/workflows/release-publish-releasenotes-twitter.yml (hunk 1 above)
Comment (romani) [reply .github/workflows/release-publish-releasenotes-twitter.yml:6]: Good

Comment (stoyanK7) [reply .github/workflows/release-publish-releasenotes-twitter.yml]: Done.

**Comment by stoyanK7** [comment]
Done.

File: .github/workflows/release-copy-github-io-to-sourceforge.yml [hunk 2]
```diff
@@ -2,8 +2,8 @@
 # GitHub Action to Copy Site to SourceForge.
 #
 #############################################################################
-name: "R: Copy Site to SourceForge"
-run-name: "R: Copy Site to SourceForge ${{ inputs.version }}"
+name: "R: Copy github.io to SourceForge"
+run-name: "R: Copy github.io to SourceForge ${{ inputs.version }}"
```
Comment (rnveach) [inline .github/workflows/release-copy-github-io-to-sourceforge.yml:6]: `github` should be `GitHub` such as https://docs.github.com/en/get-started/writing-on-github

File: .github/workflows/release-copy-github-io-to-sourceforge.yml [hunk 3]
```diff
@@ -2,8 +2,8 @@
 # GitHub Action to Copy Site to SourceForge.
```
Comment (rnveach) [inline .github/workflows/release-copy-github-io-to-sourceforge.yml:2]: Site should be changed too.

There are some other places that say `site` too. They should probably be changed or atleast reviewed.
Example:
````
  copy-site:
    name: Copy site to sourceforge for ${{ inputs.version }}
````

File: .github/workflows/release-maven-perform.yml [hunk 4]
```diff
@@ -2,8 +2,8 @@
 # GitHub Action to Run Maven Release Perform.
```
Comment (rnveach) [inline .github/workflows/release-maven-perform.yml:2]: `Maven Release Perform` => `Maven Perform`

File: .github/workflows/release-maven-prepare.yml [hunk 5]
```diff
@@ -2,8 +2,8 @@
 # GitHub Action to Run Maven Release Prepare.
```
Comment (rnveach) [inline .github/workflows/release-maven-prepare.yml:2]: same.

File: .github/workflows/release-update-github-page.yml [hunk 6]
```diff
@@ -2,8 +2,8 @@
 # GitHub Action to Publish Release Notes GitHub Page.
```
Comment (rnveach) [inline .github/workflows/release-update-github-page.yml:2]: I believe this is the same, not in synch with name.

File: .github/workflows/release-update-xdoc-with-releasenotes.yml [hunk 7]
```diff
@@ -2,8 +2,8 @@
 # GitHub Action to Push Release Notes.
```
Comment (rnveach) [inline .github/workflows/release-update-xdoc-with-releasenotes.yml:2]: Same.

**Review by rnveach — CHANGES_REQUESTED** [review]
Please note I can't put comments on lines that weren't changed and it seems to me some sub-names were not updated and they look like they need to be based on the renames we are doing. I would review all the files and verify.

File: .github/workflows/release-maven-perform.yml (hunk 4 above)
Comment (romani) [reply .github/workflows/release-maven-perform.yml:2]: Release in middle in name of plugin.

It is no maven activity, release can be done in many ways, we use this simple general plugin.

File: .github/workflows/release-copy-github-io-to-sourceforge.yml (hunk 2 above)
Comment (romani) [reply .github/workflows/release-copy-github-io-to-sourceforge.yml:6]: Ok to change. 

Or change SourceForge to sourceforge.org

File: .github/workflows/release-copy-github-io-to-sourceforge.yml (hunk 3 above)
Comment (romani) [reply .github/workflows/release-copy-github-io-to-sourceforge.yml:2]: Copy github.io to sourceforge.org

File: .github/workflows/release-maven-perform.yml (hunk 4 above)
Comment (romani) [reply .github/workflows/release-maven-perform.yml:2]: @rnveach , 
What about this 
`Run Perform goal of Maven Release plugin .`?

**Comment by rnveach** [comment]
I am fine with renames done.

I do have some comments @romani after looking at the shell files.

> release-update-xdoc-with-releasenotes.sh

This script seems to be just a wrapper for `releasenotes-gen-xdoc-push.sh`. Why don't we merge these 2 together into 1 script? I wasn't seeing it being used anywhere else.

> release-create-issues-in-other-repos.sh

This currently only creates issues in eclipse-cs and sonar. We should expand this to our others repos so we can ensure someone is updating them unless we are deprecating them. The current script doesn't cover sevntu and patch filters.

File: .github/workflows/release-update-github-page.yml (hunk 6 above)
Comment (romani) [reply .github/workflows/release-update-github-page.yml:2]: Comment is ok to be more verbose 

`Publish Release Notes to GitHub release Page`

File: .github/workflows/release-copy-github-io-to-sourceforge.yml (hunk 2 above)
Comment (stoyanK7) [reply .github/workflows/release-copy-github-io-to-sourceforge.yml:6]: Before I push, is `R: Copy github.io to sourceforge.org` okay?

File: .github/workflows/release-update-xdoc-with-releasenotes.yml (hunk 7 above)
Comment (romani) [reply .github/workflows/release-update-xdoc-with-releasenotes.yml:2]: Push release notes to releasenotes.xml file

File: .github/workflows/release-copy-github-io-to-sourceforge.yml (hunk 2 above)
Comment (rnveach) [reply .github/workflows/release-copy-github-io-to-sourceforge.yml:6]: I dunno, are we going to change `release-copy-github-io-to-sourceforge.yml` to `release-copy-github-io-to-sourceforge-org.yml`. It seems excessively specific.

I am fine with whatever @romani decides, but I think it should just be GitHub and Sourceforge` as they are the names of the sites and the actual domains shouldn't matter for just simple naming.

**Comment by romani** [comment]
We tried to be modular enough to be able to return stuff in CI and from local

https://github.com/checkstyle/checkstyle/blob/5517a72825dc75536b148ba22991e66badec2491/.ci/release-update-xdoc-with-releasenotes.sh

If no big problems, I would keep it for in such way.

File: .github/workflows/release-copy-github-io-to-sourceforge.yml (hunk 2 above)
Comment (stoyanK7) [reply .github/workflows/release-copy-github-io-to-sourceforge.yml]: Done.

File: .github/workflows/release-copy-github-io-to-sourceforge.yml (hunk 3 above)
Comment (stoyanK7) [reply .github/workflows/release-copy-github-io-to-sourceforge.yml]: Done.

File: .github/workflows/release-maven-perform.yml (hunk 4 above)
Comment (stoyanK7) [reply .github/workflows/release-maven-perform.yml]: Done.

File: .github/workflows/release-maven-prepare.yml (hunk 5 above)
Comment (stoyanK7) [reply .github/workflows/release-maven-prepare.yml]: Done.

File: .github/workflows/release-update-github-page.yml (hunk 6 above)
Comment (stoyanK7) [reply .github/workflows/release-update-github-page.yml]: Done.

File: .github/workflows/release-update-xdoc-with-releasenotes.yml (hunk 7 above)
Comment (stoyanK7) [reply .github/workflows/release-update-xdoc-with-releasenotes.yml]: Done.

**Review by rnveach — APPROVED** [review]

**Pull request merged**